From 55e773b8f8ee55b2ab12d1b269dd1988593dc02c Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@altera.com>
Date: Thu, 25 Apr 2013 09:43:57 -0500
Subject: [PATCH 237/254] tty: lcd: support tilde character on newhaven lcd

Upstream: git://git.rocketboards.org/linux-socfpga.git
 module

The NewHaven LCD module displays a right arrow instead of tilde
for an ascii 0x7e.  Load a custom font into cg ram to support tilde.

Signed-off-by: Alan Tull <atull@altera.com>
(cherry picked from commit 78de72709a21f8ef4351a92de96705d66a642c0b)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/tty/newhaven_lcd.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/tty/newhaven_lcd.c b/drivers/tty/newhaven_lcd.c
index 65d8893..4507e4f 100644
--- a/drivers/tty/newhaven_lcd.c
+++ b/drivers/tty/newhaven_lcd.c
@@ -47,10 +47,12 @@
 #define ASCII_ESC               0x1b
 #define ASCII_SPACE             0x20
 #define ASCII_BACKSLASH         0x5c
+#define ASCII_TILDE             0x7e
 
 /* The NewHaven display has 8 custom characters that are user-loadable init
    its cg ram. */
 #define CUSTOM_BACKSLASH        0x00
+#define CUSTOM_TILDE            0x01
 
 struct custom_font {
 	u8 mapping;
@@ -59,6 +61,7 @@ struct custom_font {
 
 struct custom_font custom_fonts[] = {
 	{ CUSTOM_BACKSLASH, { 0x00, 0x10, 0x08, 0x04, 0x02, 0x01, 0x00, 0x00, }, },
+	{ CUSTOM_TILDE,     { 0x00, 0x00, 0x00, 0x08, 0x15, 0x02, 0x00, 0x00, }, },
 };
 
 struct lcd {
@@ -154,6 +157,8 @@ static char lcd_translate_printable_char(char val)
 {
 	if (val == ASCII_BACKSLASH)
 		return CUSTOM_BACKSLASH;
+	else if (val == ASCII_TILDE)
+		return CUSTOM_TILDE;
 
 	return val;
 }
-- 
1.7.5.4

